File: packages/sdks/azure-native/lib/src/hybridcloud/resource_reference.dart
// ignore_for_file: unused_element, unnecessary_cast

import 'package:pulumi/pulumi.dart' as pulumi;

/// Resource reference properties.
class ResourceReference {
  /// Resource ID.
  final pulumi.Input<String>? id;

  /// Creates a new [ResourceReference].
  /// [id] Resource ID.
  ResourceReference({
    this.id,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'id': ?id,
    };
  }

  factory ResourceReference.fromMap(Map<String, dynamic> map) {
    return ResourceReference(
      id: map['id'] == null ? null : (map['id'] as String).input(),
    );
  }
}

File: pulumi-language-dart/codegen/render/resource_reference.go
package render

import (
	"fmt"
	"strings"

	"github.com/kingwill101/pulumi-dart/pulumi-language-dart/codegen/dartir"
)

func writeResourceReferenceConstructor(b *strings.Builder, declaration dartir.ResourceClass) {
	if declaration.Kind != dartir.CustomResource {
		return
	}
	metadata := resourceMetadataArguments(declaration)
	fmt.Fprintf(
		b,
		"\n  /// Creates a typed reference to an existing [%s] resource.\n"+
			"  %s.reference(String urn)\n"+
			"    : super(\n"+
			"        %s,\n"+
			"        pulumi.parseUrn(urn).urnName,\n"+
			"        const <String, pulumi.Input<dynamic>>{},\n"+
			"        pulumi.CustomResourceOptions(urn: pulumi.input(urn))%s,\n"+
			"        isResourceReference: true,\n"+
			"      )",
		declaration.Name,
		declaration.Name,
		declaration.TokenLiteral,
		metadata,
	)
	writeConstructorAssignments(b, declaration.ConstructorAssignments)
}
